lib/api/core/storage/storageEngine.js

```diff
@@ -101,18 +101,18 @@ class StorageEngine {
   _add ({ index, collection, scope='public', notify=true } = {}) {
     let modified = false;
 
-    if (! index || ! collection) {
-      return modified;
-    }
+    if (index) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the nested one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ACollapsibleIfStatements)